### ISSUE TITLE
Weird overshadowing of `this`

### ISSUE THREAD

**Issue by christianbundy** [google/traceur-compiler#1032]
Here's my code:

``` js
'use strict';

Router.configure({
  layoutTemplate: 'layout'
});

Router.onBeforeAction('loading');

Router.map(() => {
  this.route('comments', Routes.comments);
  this.route('hot', Routes.hot);
  this.route('random', Routes.random);
  this.route('recent', Routes.recent);
  this.route('root', Routes.root);
  this.route('top', Routes.top);
  this.route('user', Routes.user);

  // needs to go last so that it doesn't catch all requests
  this.route('default', Routes.default);
});
```

And here's what happens:

``` js
module.exports = function() {
  "use strict";
  'use strict';
  var $__0 = this;
  Router.configure({layoutTemplate: 'layout'});
  Router.onBeforeAction('loading');
  Router.map((function() {
    $__0.route('comments', Routes.comments);
    $__0.route('hot', Routes.hot);
    $__0.route('random', Routes.random);
    $__0.route('recent', Routes.recent);
    $__0.route('root', Routes.root);
    $__0.route('top', Routes.top);
    $__0.route('user', Routes.user);
    $__0.route('default', Routes.default);
  }));
  return {};
}.call(typeof global !== 'undefined' ? global : this);
```

Obviously the `this` in my `Router.map` isn't referring to the parent function, but it breaks my code anyway. :(
